src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlDataReader.cs

```diff
@@ -3645,16 +3683,17 @@ private bool TryReadInternal(bool setTimeout, out bool more)
                     if ((!_sharedState._dataReady) && (_stateObj.HasPendingData))
                     {
                         byte token;
-                        if (!_stateObj.TryPeekByte(out token))
+                        TdsOperationStatus debugResult = _stateObj.TryPeekByte(out token);
```

Comment:
Because the operation only exists within a DEBUG conditional block and I wanted to make sure that if code was moved around it wasn't possible to have debug and release builds somehow end up with different results. So the code in the DEBUG block is isolated from the release code.